Point MachineSet references to the official documentation in the Red Hat released CSV

jrvaldes opened the pull request:
This change links the MachineSet references in the Red Hat released CSV to
the official OpenShift docs, so that the Red Hat released operator in the
OperatorHub for OCP points to the appropriated documentation. The links to
the WMCO repository in GitHub remain as supporting information for the
community released CSV.

Ran:
```
  make bundle
```

jrvaldes commented:
/hold

depends on
- https://github.com/openshift/windows-machine-config-operator/pull/1230

jrvaldes commented:
/cherry-pick release-4.11

saifshaikh48 commented:
/lgtm

aravindhp commented on bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:
```diff
@@ -132,8 +132,8 @@ spec:
     Example MachineSet for other cloud providers:
-    - [AWS](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/machineset-aws.md)
-    - [Azure](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/machineset-azure.md)
+    - [AWS](https://docs.openshift.com/container-platform/4.11/windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.html)
```
Do we have a stance on what these links should be for the community vs Red Hat operator?

alinaryan replied:
I think we said community docs should link to our github repo, but @openshift/openshift-team-windows-containers please confirm

jrvaldes replied:
> Do we have a stance on what these links should be for the community vs Red Hat operator?

Not sure if we discussed community doc links in the past. My two cents: 
- Links for  Red Hat operator should point to official [docs](https://docs.openshift.com/container-platform/latest/windows_containers/)
- Links for the community may also point to the `latest` official docs (less maintenance). GitHub may be fine too.

mansikulkarni96 replied:
We have been pointing to github docs for community operator and official docs for RH operator. 
Is there a reason for updating these links? trying to get a context.

aravindhp replied:
>We have been pointing to github docs for community operator and official docs for RH opera

This is the correct approach. We should not link the community CSV to the official docs. It is not about maintenance but about being able to release features quicker that won't have official docs.

jrvaldes replied:
Agree

aravindhp replied:
Why are we updating this? Is this so that OpenShift docs are reflected in the Red Hat operator?

jrvaldes replied:
To use OpenShift docs in the Red Hat operator CSV

aravindhp replied:
Then please call that out. Our PRs and commits should say WHAT is being done and WHY. I am noticing quite a few commits where the WHY is missing.

jrvaldes replied:
Thanks for highlighting this.  Commit message updated.

jrvaldes committed:
[bundle] Use official docs in RedHat released CSV

This change links the MachineSet references in the Red Hat released CSV to
the official OpenShift docs, so that the Red Hat released operator in the
OperatorHub for OCP points to the appropriated documentation. The links to
the WMCO repository in GitHub remain as supporting information for the
community released CSV.

Ran: make bundle

jrvaldes commented:
/hold cancel

saifshaikh48 commented:
/lgtm

jrvaldes commented on bundle/manifests/windows-machine-config-operator.clusterserviceversion.yaml:
```diff
@@ -132,9 +132,9 @@ spec:
-    - [GCP](https://github.com/openshift/windows-machine-config-operator/blob/master/docs/machineset-gcp.md)
+    - [AWS](https://docs.openshift.com/container-platform/4.12/windows_containers/creating_windows_machinesets/creating-windows-machineset-aws.html)
+    - [Azure](https://docs.openshift.com/container-platform/4.12/windows_containers/creating_windows_machinesets/creating-windows-machineset-azure.html)
+    - [GCP](https://docs.openshift.com/container-platform/4.12/windows_containers/creating_windows_machinesets/creating-windows-machineset-gcp.html)
```
`creating-windows-machineset-gcp.html`  is proposed in https://github.com/openshift/openshift-docs/pull/49719

jrvaldes commented:
/test gcp-e2e-operator

Failed deprovisioning cluster

aravindhp commented:
/override ci/prow/gcp-e2e-operator

sebsoto commented:
/lgtm
